pkg/capi/deploy.go

```diff
@@ -256,3 +259,25 @@ func (clusterAPI *Manager) DeployCluster(ctx context.Context, clusterName string
 
 	return deployedCluster, nil
 }
+
+// DestroyCluster deletes cluster.
+func (clusterAPI *Manager) DestroyCluster(ctx context.Context, name, namespace, version string) error {
+	cluster := &unstructured.Unstructured{}
+	cluster.SetName(name)
+	cluster.SetNamespace(namespace)
+	cluster.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "cluster.x-k8s.io",
+		Kind:    "Cluster",
+		Version: version,
+	})
+
+	if err := clusterAPI.runtimeClient.Delete(ctx, cluster); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
+
+		return err
+	}
```

Comment:
do we want to wait for the cluster to be actually deleted here?

depends on the usecase, but might be good measure to make sure AWS provider has a chance to clean up everything

(basically a for loop waiting for Get to return 'NotFound')

Reply:
right, good point.